homeassistant/components/xbox/base_sensor.py

```diff
@@ -44,7 +44,13 @@ def entity_picture(self) -> str:
         if not self.data:
             return None
 
-        return self.data.display_pic.replace("&mode=Padding", "")
+        # Xbox sometimes returns a domain that uses a wrong certificate which creates issues
+        # with loading the image.
+        # The correct domain is images-eds-ssl which can just be replaced
+        # to point to the correct image, with the correct domain and certificate
+        return self.data.display_pic.replace("&mode=Padding", "").replace(
+            "images-eds.xboxlive.com", "images-eds-ssl.xboxlive.com"
+        )
```

Comment:
Even if it ends up more lines, please use proper url parsing instead of string routines.

Reply:
I'm sorry I don't understand what you mean with proper url parsing? Is that another way to replace the domain? 
All we are trying to achieve here is inserting "-ssl" into the domain.

Reply:
https://stackoverflow.com/a/47870979

Reply:
Or as also mentioned in above thread: http://mpwang.github.io/2018/12/31/python-replace-url-host/

Reply:
Or simpker Use furl's url object. 
```
from furl import URL

print(str(URL(original).with_host("new host"))
```

Reply:
I'm making some offline progress. I have succesfully setup a dev environment. It does take some time for me to grasp everything. I hope to get the integration tested tomorrow, right now i think i'm hitting an api limit. 

Just wanted to update so you know i'm working on it.